proto/internal/uber/cadence/matching/v1/service.proto

```diff
@@ -97,10 +97,16 @@ service MatchingAPI {
   rpc RefreshTaskListPartitionConfig(RefreshTaskListPartitionConfigRequest) returns (RefreshTaskListPartitionConfigResponse);
 }
 
+message TaskListPartition {
+  repeated string isolation_groups = 1;
+}
+
 message TaskListPartitionConfig {
   int64 version = 1;
-  int32 num_read_partitions = 2;
-  int32 num_write_partitions = 3;
+  int32 num_read_partitions = 2 [deprecated = true];
```

Comment:
We can mark the message deprecated and use the one publicly exposed.

Reply:
Agreed. I'd prefer to do it in a subsequent change though. I started working on it but it's another decent sized refactoring to do so safely.